### DISCUSSION TITLE
How to make  the component Icon appear on the same line.

### DISCUSSION THREAD

**Discussion by vhtmui** [withastro/starlight#1791] [Q&A] [answered]
Im a beginner, and i can't make the **Icon** on the same line, they display like this.
```
1<Icon name="star" size="1rem" />2<Icon name="star" size="1rem" />3<Icon name="star" size="1rem" />
```
![1714235288451](https://github.com/withastro/starlight/assets/46163279/1e068534-38b5-451f-ab87-2ce20b48b23d)

**Comment by HiDeoo** [comment, marked as answer]
:wave: You should be able to target the specific icons you wish to display inline by passing a custom CSS class to the icon component:

```md
text <Icon name="star" class="inline-icon />
```

Then, you would need to [add custom CSS styles](https://starlight.astro.build/guides/css-and-tailwind/#custom-css-styles) for this new class by specifying a path to a CSS file in `astro.config.mjs`:

```diff
export default defineConfig({
  integrations: [
    starlight({
      title: 'Your docs',
      customCss: [
+        // Relative path to your custom CSS file.
+        './src/styles/custom.css',
      ],
    }),
  ],
});
```

And in `src/styles/custom.css`, add a style for the `.inline-icon` class to display the icon inline:

```css
.sl-markdown-content .inline-icon {
  display: inline;
}
```

**Reply by vhtmui** [reply]
Seems it doesn't work, in `mdx` file :
```
text <Icon name="star" class='inline-icon' /> text
```
In custom css :
```
.inline-icon {
	display: inline;
	color: red;
}
```
and it looks like this :
![1714240046723](https://github.com/withastro/starlight/assets/46163279/2dd5b927-38a7-45ae-b69c-19425a8d5c51)

**Reply by HiDeoo** [reply]
Ah indeed, not something I can test right now as I'm on a phone, but the default Starlight Markdown style should enforce that.

I would guess that switching to `.sl-markdown-content .inline-icon` for the selector in your custom CSS file would do the trick (I'll edit my initial reply).

**Reply by vhtmui** [reply]
it do work now, thank u 👍
